platform/ios/src/MGLMapboxEvents.m

```diff
@@ -433,51 +434,63 @@ - (MGLMapboxEventAttributes *)fullyFormedEventForEvent:(NSString *)event withAtt
     return nil;
 }
 
+- (void)updateDictionary:(NSMutableDictionary *)dictionary withKey:(NSString *)key value:(id)value {
+    if (key && value) {
+        [dictionary setValue:value forKey:key];
+    }
+}
+
 - (MGLMapboxEventAttributes *)locationEventWithAttributes:(MGLMapboxEventAttributes *)attributeDictionary {
-    MGLMutableMapboxEventAttributes *attributes = [@{MGLEventKeyEvent: MGLEventTypeLocation,
-                                                     MGLEventKeySource: MGLEventSource,
-                                                     MGLEventKeySessionId: self.instanceID,
-                                                     MGLEventKeyOperatingSystem: self.data.iOSVersion} mutableCopy];
+    MGLMutableMapboxEventAttributes *attributes = [NSMutableDictionary dictionary];
+    [self updateDictionary:attributes withKey:MGLEventKeyEvent value:MGLEventTypeLocation];
+    [self updateDictionary:attributes withKey:MGLEventKeySource value:MGLEventSource];
+    [self updateDictionary:attributes withKey:MGLEventKeySessionId value:self.instanceID];
```

Comment:
Since this is already a mutable dictionary and the keys are constant, it would be just as easy to say e.g. `attributes[MGLEventKeySessionId] = self.instanceID`.